packages/twenty-front/src/modules/object-record/spreadsheet-import/hooks/useBuildSpreadSheetImportFields.ts

```diff
@@ -0,0 +1,262 @@
+    uniqueConstraintField,
+    uniqueConstraintType,
+  }: {
+    fieldMetadataItem: FieldMetadataItem;
+    uniqueConstraintField: FieldMetadataItem;
+    uniqueConstraintType: CompositeFieldType;
+  }) => {
+    const spreadsheetImportFields: SpreadsheetImportField[] = [];
+
+    SETTINGS_COMPOSITE_FIELD_TYPE_CONFIGS[
+      uniqueConstraintType
+    ].subFields.forEach(
+      ({ subFieldName, isImportable, isIncludedInUniqueConstraint }) => {
+        if (!isImportable || !isIncludedInUniqueConstraint) return;
+
+        spreadsheetImportFields.push(
+          createBaseField(fieldMetadataItem, {
+            label: getRelationConnectSubFieldLabel(
+              fieldMetadataItem,
+              uniqueConstraintField,
+              subFieldName,
+            ),
+            key: getRelationConnectSubFieldKey(
+              fieldMetadataItem,
+              uniqueConstraintField,
+              subFieldName,
+            ),
+            fieldValidationDefinitions:
+              getSpreadSheetFieldValidationDefinitions(
+                uniqueConstraintField.type,
+                uniqueConstraintField.name,
+                subFieldName,
+              ),
+            isNestedField: true,
+            isCompositeSubField: true,
+            compositeSubFieldKey: subFieldName,
+            uniqueFieldMetadataItem: uniqueConstraintField,
+            isRelationConnectField: true,
+          }),
+        );
+      },
+    );
+
+    return spreadsheetImportFields;
+  };
+
+  const handleSelectField = (
+    fieldMetadataItem: FieldMetadataItem,
+    isMulti = false,
+    subFieldOverrides?: Record<string, any>,
+  ) =>
+    createBaseField(fieldMetadataItem, {
+      fieldType: {
+        type: isMulti ? 'multiSelect' : 'select',
+        options:
+          fieldMetadataItem.options?.map((option) => ({
+            label: option.label,
+            value: option.value,
+            color: option.color,
+          })) || [],
+        ...(isDefined(subFieldOverrides) ? subFieldOverrides : {}),
+      },
+      fieldValidationDefinitions: getSpreadSheetFieldValidationDefinitions(
+        fieldMetadataItem.type,
+        `${fieldMetadataItem.label} (ID)`,
+      ),
+    });
+
+  const handleRelationField = (fieldMetadataItem: FieldMetadataItem) => {
+    const spreadsheetImportFields: SpreadsheetImportField[] = [];
+
+    const isManyToOneRelation =
+      fieldMetadataItem.relation?.type === RelationType.MANY_TO_ONE;
+
+    const targetObjectMetadataItem = objectMetadataItems?.find(
+      (objectMetadataItem) =>
+        objectMetadataItem.id ===
+        fieldMetadataItem.relation?.targetObjectMetadata.id,
+    );
+
+    if (isManyToOneRelation && isDefined(targetObjectMetadataItem)) {
+      const uniqueConstraintFields = getUniqueConstraintsFields<
+        FieldMetadataItem,
+        ObjectMetadataItem
+      >(targetObjectMetadataItem);
+
+      for (const uniqueConstraintField of uniqueConstraintFields.flat()) {
```

Comment:
Add a comment to explain : as of today the composite indexes on core schema are not possible, but may happen in the future